### DISCUSSION TITLE
2 Error messages IPXE UEFI Boot with Fog

### DISCUSSION THREAD

**Discussion by JeremyH-NSP** [ipxe/ipxe#743] [General]
Hello,
I am receiving two messages when I try to image or register a computer with FOG.  I can IPXE Boot using UEFI just fine.  I get to the FOG menu.  I choose either to register or deploy the image.  If I choose register I get one of the two error messages depending on the brand of computer I am using.  

If I am trying to capture an image or deploy an image using an HP computer I receive the following error message.
**Could not boot: Exec format error (https://ipxe.org/2e008081)**

If I am trying to capture an image or deploy an image using a Lenovo mini desktop computer I receive the following error message. 
 **IFog: Error 0x7f048283 (http://ipxe.org/7f048283)**

I have updated to the latest version of IPXE, but I am still receiving the errors.  I can capture and register computers just fine using IPXE and Fog when booting with Legacy mode.  The errors only happen using UEFI PXIE booting.  I would like to get this working.  Please let me know what I can do to fix it.

**Comment by stappersg** [comment]
> I would like to get this working. Please let me know what I can do to fix it.

Make the problem reproducable  and demostrate that the iPXE-Fog-combinations works for non uEFI systems.

That will require **two** `qemu` command lines. One with a _QEMU BIOS system_ and the other with a _QEMU uEFI system_.  At https://lists.ipxe.org/pipermail/ipxe-devel/2022-August/007581.html is this
```text
  sudo qemu-system-x86_64 -boot d -cdrom bin/ipxe.iso \
     -device rtl8139,netdev=rope0,mac=52:54:00:4e:69:63 \
     -netdev bridge,id=rope0,br=ltbr0
```
reported as a working iPXE system.

**Comment by NiKiZe** [comment]
You can read the error pages to get suggestions.
Otherwise you need to explain your issue better.

What image are you deploying? How does the iPXE script look? Which image is booted? Do you understand what the linked error pages say?

This was moved to discussion since it is unlikely to be an issue with iPXE, and more likely an issue with your configuration / script / images.

**Comment by stappersg** [comment]
Link to a relevant _FOG Project_ webpage:  https://wiki.fogproject.org/wiki/index.php?title=IPXE

**Comment by chainofhonor** [comment]
Try to use Windows Server 2022 IIS to provide www service, set MIME type and add ISO suffix

I use openwrt to download pe files. Everything is normal in BIOS mode, but it cannot be started in UEFI mode

That's the command   **sanboot http://192.168.1.1/pe/pe.iso**

But if I change the URL to IIS, it can start normally

I guess it may be a MIME type problem, but strangely, the BIOS mode is OK

**Reply by NiKiZe** [reply]
Your message is not related to this discussion, you did however create #753, so keep it there ;)
